plugins/inputs/ravendb/ravendb.go

```diff
@@ -0,0 +1,443 @@
+			}
+
+			if indexesResponse.PublicServerUrl != nil {
+				tags["public_server_url"] = *indexesResponse.PublicServerUrl
+			}
+
+			fields := map[string]interface{}{
+				"errors":                          indexResponse.Errors,
+				"is_invalid":                      indexResponse.IsInvalid,
+				"lock_mode":                       indexResponse.LockMode,
+				"mapped_per_sec":                  indexResponse.MappedPerSec,
+				"priority":                        indexResponse.Priority,
+				"reduced_per_sec":                 indexResponse.ReducedPerSec,
+				"state":                           indexResponse.State,
+				"status":                          indexResponse.Status,
+				"time_since_last_indexing_in_sec": indexResponse.TimeSinceLastIndexingInSec,
+				"time_since_last_query_in_sec":    indexResponse.TimeSinceLastQueryInSec,
+				"type":                            indexResponse.Type,
+			}
+
+			acc.AddFields("ravendb_indexes", fields, tags)
+		}
+	}
+}
+
+func (r *RavenDB) gatherCollections(acc telegraf.Accumulator) {
+	collectionsResponse := &collectionsMetricResponse{}
+
+	err := r.requestJSON(r.requestUrlCollection, &collectionsResponse)
+	if err != nil {
+		acc.AddError(err)
+		return
+	}
+
+	for _, perDbCollectionMetrics := range collectionsResponse.Results {
+		for _, collectionMetrics := range perDbCollectionMetrics.Collections {
+			tags := map[string]string{
+				"collection_name": collectionMetrics.CollectionName,
+				"database_name":   perDbCollectionMetrics.DatabaseName,
+				"node_tag":        collectionsResponse.NodeTag,
+				"url":             r.URL,
+			}
+
+			if collectionsResponse.PublicServerUrl != nil {
+				tags["public_server_url"] = *collectionsResponse.PublicServerUrl
+			}
+
+			fields := map[string]interface{}{
+				"documents_count":          collectionMetrics.DocumentsCount,
+				"documents_size_in_bytes":  collectionMetrics.DocumentsSizeInBytes,
+				"revisions_size_in_bytes":  collectionMetrics.RevisionsSizeInBytes,
+				"tombstones_size_in_bytes": collectionMetrics.TombstonesSizeInBytes,
+				"total_size_in_bytes":      collectionMetrics.TotalSizeInBytes,
+			}
+
+			acc.AddFields("ravendb_collections", fields, tags)
+		}
+	}
+}
+
+func prepareDbNamesUrlPart(dbNames []string) string {
+	if len(dbNames) == 0 {
+		return ""
+	}
+	result := "?" + dbNames[0]
+	for _, db := range dbNames[1:] {
+		result += "&name=" + url.QueryEscape(db)
+	}
+
+	return result
+}
+
+func (r *RavenDB) Init() error {
+	if r.URL == "" {
+		r.URL = defaultURL
+	}
+
+	r.requestUrlServer = fmt.Sprintf("%s%s", r.URL, "/admin/monitoring/v1/server")
+	r.requestUrlDatabases = fmt.Sprintf("%s%s", r.URL, "/admin/monitoring/v1/databases"+prepareDbNamesUrlPart(r.DbStatsDbs))
+	r.requestUrlIndexes = fmt.Sprintf("%s%s", r.URL, "/admin/monitoring/v1/indexes"+prepareDbNamesUrlPart(r.IndexStatsDbs))
+	r.requestUrlCollection = fmt.Sprintf("%s%s", r.URL, "/admin/monitoring/v1/collections"+prepareDbNamesUrlPart(r.IndexStatsDbs))
```

Comment:
I think you can even get rid of the `fmt.Sprintf()`
```suggestion
	r.requestUrlServer = r.URL + "/admin/monitoring/v1/server"
	r.requestUrlDatabases = r.URL + "/admin/monitoring/v1/databases" + prepareDbNamesUrlPart(r.DbStatsDbs)
	r.requestUrlIndexes = r.URL + "/admin/monitoring/v1/indexes" + prepareDbNamesUrlPart(r.IndexStatsDbs)
	r.requestUrlCollection = r.URL + "/admin/monitoring/v1/collections"+prepareDbNamesUrlPart(r.IndexStatsDbs)
```

Reply:
done